backendCore/dl/dl_model_parser.py

```diff
@@ -0,0 +1,84 @@
+from sklearn.tree import *
+from sklearn.ensemble import *
+from sklearn.linear_model import *
+from sklearn.svm import *
+from sklearn.naive_bayes import *
+import torch
+import torch.nn as nn
+from torch.autograd import Variable
+import torchvision.transforms as transforms
+
+
+def parse_deep_user_architecture(user_model):
+    """
+    new:
+    Each layer in the user_model has this format:
+    {"value": "LAYER_NAME", "parameters": [PARAMETER_1, PARAMETER_2, ...]}
+
+    Eg: [{"value": "LINEAR", "parameters": [50, 10]}] should become [nn.Linear(in_features=50, out_features=10)]
+
+    Args:
+        user_model (list): string form where you have the model name and the values for each parameter
+    Returns:
+        parsed_modules (list): list of nn.Module objects parsed
+    ----------------------------------------------------------------------------------------------
+    old:
+    Given a list of user_models in a certain order,
+    each layer should contain name and the values of each
+    parameter
+
+    Eg: ["Linear(in_features=50, out_features=10)"] should become [nn.Linear(in_features=50, out_features=10)]
+
+    ["nn.Linear(in_features=50, out_features=10)", "nn.Conv2d(in_channels=16, out_channels=33, kernel_size=3, stride=2)"] should become
+    [nn.Linear(in_features=50, out_features=10), nn.Conv2d(in_channels=16, out_channels=33, kernel_size=3, stride=2)]
+
+    Args:
+        user_model (list): string form where you have the model name and the values for each parameter
+    Returns:
+        parsed_modules (list): list of nn.Module objects parsed
+    """
+
+    # JSON file will contain name of model along with what hyperparameters need to be specified
+    parsed_modules = []
+    for element in user_model:
+        if element["value"] == "LINEAR":
```

Comment:
If we are accounting for other layers like conv2d, gaussianblur, etc in this parser function, do you think we can potetnially generalize this implementation so that we don't have to keep adding if/elif/elif... chains and cause maintainability issues?